acra/configuring-maintaining/general-configuration/acra-authmanager.md

```diff
@@ -0,0 +1,131 @@
+
+* `--file=<filepath>`
+  
+    Path to encrypted file where stored authentication data
+    Default is `configs/auth.keys`
+
+* `--user=<username>`
+
+    Name of user that will be added/removed/updated. This flag is required.
+  
+* `--password=<password>`
+
+    User's password that will be used for basic authentication. Cannot be empty for `--set` operation and unused for 
+    `--remove` operation.
+  
+* `--remove`
+  
+    Flag to remove user specified by `--user=<user>` flag. 
+  
+* `--set`
+
+    Flag to add/update password for user specified by `--user=<user>` flag. Password cannot be empty.
+
+### Configuration files
+
+* `--config_file=<filename>`
+
+  Path to YAML configuration file.
+
+* `--dump_config`
+
+  Dump configuration to `configs/acra-addzone.yaml`.
+
+* `--generate_markdown_args_table`
+
+  Generate markdown file with text description of all flags.
+  Output file is `configs/markdown_acra-addzone.md`.
+  Works in pair with `--dump_config`.
+
+### Logging
+
+* `-d`
+
+  Log to stderr all `DEBUG`, `INFO`, `WARNING` and `ERROR` logs.
+
+
+### Storage destination
+
+#### Filesystem
+
+* `--keys_dir=<path>`
+
+  Path to keystore directory.
+
+  Default is `.acrakeys`.
+
+### HashiCorp Vault
+
+`acra-authmanager` can read `ACRA_MASTER_KEY` from HashiCorp Vault instead of environment variable.
+
+* `--vault_connection_api_string=<url>`
+
+  Connection string (like `https://example.com:8200`) for connecting to HashiCorp Vault.
+  If not specified, `ACRA_MASTER_KEY` environment variable will be used.
+
+* `--vault_secrets_path=<kv-path>`
+
+  Path to KV Secrets directory in Vault used to store `ACRA_MASTER_KEY`.
+  Default is `secret/`.
+
+* `--vault_tls_ca_path=<path>`
+
+  Path to CA certificate bundle to use for HashiCorp Vault certificate validation.
+
+  If not specified, use root certificates configured in system.
+
+* `--vault_tls_client_cert=<path>`
+
+  Path to client TLS certificate used to connect to HashiCorp Vault.
+
+  If not specified, don't send client certificate.
+
+* `--vault_tls_client_key=<path>`
+
+  Path to the private key of the client TLS certificate used to connect to HashiCorp Vault.
+
+  If not specified, don't send client certificate.
+
+* `--vault_tls_transport_enable={true|false}`
+
+  Use TLS to encrypt transport with HashiCorp Vault.
+  Default is `false`.
+
+## Auth file
+
+`acra-authmanager` updates encrypted file where stored rows in format: 
+```
+<user>:<salt>:<hash_function_parameters>:<hash(salt, password, parameters)>
+<user>:<salt>:<hash_function_parameters>:<hash(salt, password, parameters)>
+<user>:<salt>:<hash_function_parameters>:<hash(salt, password, parameters)>
+``` 
+Each row is separate entry related to distinct user. 
+
+Every password hashed using [Argon2 hash function](https://en.wikipedia.org/wiki/Argon2). You can find current parameter
+values used for hash function in [Acra source code](https://github.com/cossacklabs/acra/blob/master/cmd/constants.go#L34).
```

Comment:
This link may become a bit outdated at some point.
Another possible solution would be using search link like [this one](https://github.com/cossacklabs/acra/search?q=DefaultWebConfigAuthArgon2Length), although it also points to other usages of the constant.

Reply:
changed to link with specified release number that will not be changed